### PR TITLE
Point of sale : Show discount when box ticked on pricelist

### PR DESCRIPTION
**Pull request by qle-odoo**
Many users are confused when using pricelists on the PoS : 
When they tick the box "Show discount" on the pricelist, it has no impact when the pricelist is applied on the PoS. 
They often report a ticket thinking it's a bug. 

It can be useful to show the discount on the PoS when applying a pricelist, when users benefit from a preferred price, 
the vendor often wants to show it to the customer. 

Task : https://www.odoo.com/web#id=1853454&action=333&active_id=1251&model=project.task&view_type=form&menu_id=4720

### CODE REVIEW

File: addons/point_of_sale/static/src/xml/pos.xml [hunk 1]
```diff
@@ -1262,7 +1262,7 @@
                 <t t-if="line.get_product().tracking!=='none'">
                     <i class="oe_link_icon fa fa-list oe_icon line-lot-icon oe_green" t-if="line.has_valid_product_lot()"
                        aria-label="Valid product lot" role="img" title="Valid product lot"/>
-                    <i class="oe_link_icon fa fa-list oe_icon line-lot-icon oe_red" t-if="not(line.has_valid_product_lot())"
+                    <i class="oe_link_icon fa fa-list oe_icon line-lot-icon oe_red" t-if="!(line.has_valid_product_lot())"
```
Comment (jorenvo) [inline addons/point_of_sale/static/src/xml/pos.xml]: Good catch :)

File: addons/point_of_sale/static/src/xml/pos.xml [hunk 2]
```diff
@@ -1277,7 +1277,16 @@
                         </em>
                         <t t-esc="line.get_unit().name" />
                         at
-                        <t t-esc="widget.format_currency(line.get_unit_display_price(),'Product Price')" />
+                        <t t-if="line.display_discount_policy() == 'without_discount' &amp;&amp; 
+                            line.get_unit_display_price() != line.get_list_price()">
+                            <strike>
+                                <t t-esc="widget.format_currency(line.get_list_price(),'Product List Price')" />
```
Comment (jorenvo) [inline addons/point_of_sale/static/src/xml/pos.xml]: The second argument to `format_currency` should be the name of a decimal precision. `Product List Price` does not exist, I think it should still be `Product Price` like in the the other calls.

File: addons/point_of_sale/static/src/js/models.js [hunk 3]
```diff
@@ -1757,6 +1760,12 @@ exports.Orderline = Backbone.Model.extend({
             "taxDetails": taxdetail,
         };
     },
+    display_discount_policy: function(){
+        return this.product.get_discount_policy(this.order.pricelist);
```
Comment (jorenvo) [inline addons/point_of_sale/static/src/js/models.js]: I would remove the `get_discount_policy` function on product and immediately return the value here. Currently the `get_discount_policy` on product doesn't actually use `this` so it's a bit weird to have it on that class.

File: addons/point_of_sale/static/src/js/models.js [hunk 4]
```diff
@@ -1285,6 +1285,9 @@ exports.Product = Backbone.Model.extend({
         // because it would cause inconsistencies with the backend for
         // pricelist that have base == 'pricelist'.
         return price;
+    }, 
+    get_discount_policy: function(pricelist){
+        return pricelist.discount_policy;
```
Comment (jorenvo) [inline addons/point_of_sale/static/src/js/models.js]: Remove this, see comment below.

File: addons/point_of_sale/static/src/xml/pos.xml [hunk 5]
```diff
@@ -1465,6 +1474,16 @@
                 <tr t-foreach="orderlines" t-as="orderline">
                     <td>
                         <t t-esc="orderline.get_product().display_name"/>
+                        <t t-if="orderline.display_discount_policy() == 'without_discount' &amp;&amp; 
+                            orderline.get_unit_display_price() != orderline.get_list_price()">
+                            <div class="pos-disc-font">
+                                <strike>
+                                    <t t-esc="widget.format_currency(orderline.get_list_price(),'Product List Price')"/>
```
Comment (jorenvo) [inline addons/point_of_sale/static/src/xml/pos.xml]: Same `Product Price` comment as above.

File: addons/sale/models/product_pricelist.py [hunk 6]
```diff
@@ -1,13 +0,0 @@
...
-from odoo import fields, models
-
-
-class ProductPricelist(models.Model):
-    _inherit = 'product.pricelist'
-
-    discount_policy = fields.Selection([
-        ('with_discount', 'Discount included in the price'),
-        ('without_discount', 'Show public price & discount to the customer')],
-        default='with_discount')
```
[the remark this answers is not in the document]
Comment (mart-e) [reply addons/sale/models/product_pricelist.py]: discussed and we agreed

**Comment by mart-e** [comment]
Applied changes but still need to compute the price before discount to display the difference on the lines

**Comment by switch87** [comment]
closed by https://github.com/odoo/odoo/pull/27299

**Pull request closed without merging**